theano/scan_module/tests/test_scan.py

```diff
@@ -4983,12 +4983,34 @@ def loss_ti(ti, sum_ti, mi, W):
                 )
 
         loss = result_mi[-1]
-        grad = tensor.grad(loss, W_flat)
+        tensor.grad(loss, W_flat)
     finally:
         theano.config.compute_test_value = old_compute_test_val
         theano.config.exception_verbosity = old_exception_verbosity
 
 
+def test_compute_test_value_grad_cast():
+    # Test for test values when variables have to be casted
+    # Reported by Daniel Renshaw at
+    # https://groups.google.com/d/topic/theano-users/o4jK9xDe5WI/discussion
+    backup = theano.config.compute_test_value
+    theano.config.compute_test_value = 'raise'
+    try:
+        h = tensor.matrix('h')
```

Comment:
This test fail when floatX=float32. Is that expected? when floatX=float64, everything in this tests is in float64. So I don't understand how this test can test the change in this PR.

Do I miss something? Can we just use dmatrix to fix this test?